src/str.c

```diff
@@ -1270,24 +1270,28 @@ char *pr_str_get_word(char **cp, int flags) {
     }
   }
 
-  if (**cp == '\"') {
-    quote_mode++;
-    (*cp)++;
+  if (!(flags & PR_STR_FL_IGNORE_QUOTES)) {
```

Comment:
This new `IGNORE_QUOTES` flag, and its use in `make_ftp_cmd`, comprise the heart of the fix here.